src/NzbDrone.Core/Organizer/FileNameBuilder.cs

```diff
@@ -279,6 +280,16 @@ private void AddArtistTokens(Dictionary<string, Func<TokenMatch, string>> tokenH
             }
         }
 
+        private void AddTrackArtistTokens(Dictionary<string, Func<TokenMatch, string>> tokenHandlers, List<Track> tracks)
+        {
+            var artists = tracks.Select(t => t.ArtistMetadata.Value)
+                .DistinctBy(a => a.Name)
+                .ToList();
+
+            tokenHandlers["{Track ArtistName}"] = m => string.Join("+", artists.Select(a => a.Name));
```

Comment:
Do we want to list these all and join by "+" or do we just want the first? should we have different tokens? @ta264 thoughts?

Reply:
@Qstick What does it mean if there's multiple tracks in the list? I copied the `+` joining based on what `{Track Title}` does but I didn't quite understand why it's done.

Reply:
Lidarr is like Sonarr in that it supports 1 file = multiple tracks. So a particular file can be associated with multiple tracks.. Ex. Single file album.

Reply:
I'll update it to just use the first one.